File: LeetCode-Programs/Last Stone Weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        while len(stones)>1:
            stones.sort()
            
            x=stones[len(stones)-2]
            y=stones[len(stones)-1]
            if x==y:
                stones.pop()
                stones.pop()
            else:
                stones.pop()
                stones.pop()
                stones.append(y-x)
        if len(stones)==0:
            return 0
        else:
            return stones[0]
            

File: LeetCode-Programs/Last Stone Weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        while(stones.size()>1){
            sort(stones.begin(), stones.end());
            int x=stones[stones.size()-2];
            int y=stones[stones.size()-1];
        
            if(x==y){
                stones.pop_back();
                stones.pop_back();
            }
            else{
                stones.pop_back();
                stones.pop_back();
                stones.push_back(y-x);
            }
        }
        if(stones.size()==0)
            return 0;
        else
            return stones[0];
    }
};

